src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -3033,6 +3052,326 @@ - (void)test032_MTRPathClassesEncoding
     XCTAssertEqualObjects(originalCommandPath, decodedCommandPath);
 }
 
+// Helper API to test if changes in an attribute with a path specified by endpointId, clusterId and attributeId affect
+// device configuration changed for a given MTRDevice. This API creates a fake attribute report for the given attribute path
+// and injects it into MTRDevice to exercise and test the delegate's device configuration changed callback.
++ (void)runTestDeviceConfigurationChangedForAttribute:(MTRAttributeIDType)attributeId
+                                            clusterId:(MTRClusterIDType)clusterId
+                                           endpointId:(NSNumber *)endpointId
+                                               device:(MTRDevice *)device
+                                             delegate:(MTRDeviceTestDelegate *)delegate
+                                          dataVersion:(NSNumber *)dataVersion
+                                      attributeReport:(NSArray<NSDictionary<NSString *, id> *> *)attributeReport
+                                             testcase:(XCTestCase *)testcase
+{
+    XCTAssertNotNil(endpointId);
+    XCTAssertNotNil(device);
+    XCTAssertNotNil(delegate);
+    XCTAssertNotNil(dataVersion);
+    XCTAssertNotNil(attributeReport);
+    XCTAssertNotNil(testcase);
+
+    XCTestExpectation * gotAttributeReportExpectation = [testcase expectationWithDescription:@"Attribute report has been received"];
+    XCTestExpectation * gotAttributeReportEndExpectation = [testcase expectationWithDescription:@"Attribute report has ended"];
+    XCTestExpectation * deviceConfigurationChangedExpectation = [testcase expectationWithDescription:@"Device configuration changed was receieved"];
+
+    __block unsigned attributeReportsReceived = 0;
+    __block NSArray<NSDictionary<NSString *, id> *> * testDataValue = nil;
+
+    for (NSDictionary<NSString *, id> * attributeDict in attributeReport) {
+        MTRAttributePath * attributePath = attributeDict[MTRAttributePathKey];
+        if (attributePath.attribute.unsignedLongLongValue == attributeId) {
+            testDataValue = attributeDict[MTRDataKey][MTRValueKey];
```

Comment:
This doesn't make sense to me.  How do we know `attributeDict[MTRDataKey][MTRValueKey]` is an array?  In general I would expect it to _not_ be an array...